library/src/main/java/com/github/windsekirun/naraeimagepicker/activity/NaraePickerActivity.kt

```diff
@@ -43,12 +46,47 @@ class NaraePickerActivity : AppCompatActivity() {
         SelectedItem.setLimits(PickerSet.getSettingItem().pickLimit)
         requestFileViewMode = PickerSet.getSettingItem().viewMode == ViewMode.FileView
 
-        RPermission.instance.checkPermission(this, arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE,
-                Manifest.permission.WRITE_EXTERNAL_STORAGE)) { integer, _ ->
-            if (integer == RPermission.PERMISSION_GRANTED) {
+        checkPermission()
+    }
+
+    private fun checkPermission() {
+        val storageReadPermission = ContextCompat.checkSelfPermission(
+                this, Manifest.permission.READ_EXTERNAL_STORAGE
+        ) == PackageManager.PERMISSION_GRANTED
+        when {
```

Comment:
In [Coding conventions in Kotlin](https://kotlinlang.org/docs/reference/coding-conventions.html#if-versus-when), we can prefer using `when` keyword **if there are three or more options**. So change `when` to `if-else` keyword.

```
if (storageReadPermission) {
            initFragment(if (requestFileViewMode) FragmentMode.All else FragmentMode.Album)
        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            requestPermissions(
                    arrayOf(
                            Manifest.permission.READ_EXTERNAL_STORAGE,
                            Manifest.permission.WRITE_EXTERNAL_STORAGE
                    ), RC_READ_STORAGE
            )
        }
```